lib/updateVersion.js

```diff
@@ -0,0 +1,632 @@
+            for (var materialValueId in materialValues) {
+                if (materialValues.hasOwnProperty(materialValueId)) {
+                    var materialValue = materialValues[materialValueId];
+                    if (!Array.isArray(materialValue)) {
+                        materialValues[materialValueId] = [materialValue];
+                    }
+                }
+            }
+        }
+    }
+}
+
+function requireAttributeSetIndex(gltf) {
+    var meshes = gltf.meshes;
+    for (var meshId in meshes) {
+        if (meshes.hasOwnProperty(meshId)) {
+            var mesh = meshes[meshId];
+            var primitives = mesh.primitives;
+            if (defined(primitives)) {
+                var primitivesLength = primitives.length;
+                for (var i = 0; i < primitivesLength; i++) {
+                    var primitive = primitives[i];
+                    var attributes = primitive.attributes;
+                    if (defined(attributes)) {
+                        var semantics = Object.keys(attributes);
+                        var semanticsLength = semantics.length;
+                        for (var j = 0; j < semanticsLength; j++) {
+                            var semantic = semantics[j];
+                            if (semantic === 'TEXCOORD') {
+                                attributes.TEXCOORD_0 = attributes[semantic];
+                                delete attributes[semantic];
+                            }
+                            if (semantic === 'COLOR') {
+                                attributes.COLOR_0 = attributes[semantic];
+                                delete attributes[semantic];
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    var techniques = gltf.techniques;
+    for (var techniqueId in techniques) {
+        if (techniques.hasOwnProperty(techniqueId)) {
+            var technique = techniques[techniqueId];
+            var techniqueParameters = technique.parameters;
+            for (var techniqueParameterId in techniqueParameters) {
+                if (techniqueParameters.hasOwnProperty(techniqueParameterId)) {
+                    var techniqueParameter = techniqueParameters[techniqueParameterId];
+                    var techniqueParameterSemantic = techniqueParameter.semantic;
+                    if (defined(techniqueParameterSemantic)) {
+                        if (techniqueParameterSemantic === 'TEXCOORD') {
+                            techniqueParameter.semantic = 'TEXCOORD_0';
+                        } else if (techniqueParameterSemantic === 'COLOR') {
+                            techniqueParameter.semantic = 'COLOR_0';
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+var knownSemantics = {
+    POSITION: true,
+    NORMAL: true,
+    TEXCOORD: true,
+    COLOR: true,
+    JOINT: true,
+    WEIGHT: true,
+    BATCHID: true
```

Comment:
`BATCHID` is part of 3D Tiles, not glTF, so the 3D Tiles spec will need to name this `_BATCHID` (please update the spec if it isn't already), and then we can have backwards compatibility loading in Cesium with a warning (also please open PR if this isn't done).